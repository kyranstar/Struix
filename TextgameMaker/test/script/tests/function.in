def twicePlusOne(n)
  return (n * 2) + 1;
end 

def times3(a)
  return a * 3;
end

def printFunc()
  println("sup");
end

def invert(input)
  return not input;
end

def add(first, second)
	return first + second;
end

println(twicePlusOne(5));
printFunc();
println(times3("Wow"));
println(invert(true));
println(add(3,6));

// functions
def twice(n)
  temp is (n * 2); 
  return temp; 
end

def squared(n) 
  return n*n; 
end

def squaredAndTwice(n)
  return twice(squared(n));
end

assert(squared(666) equals 666^2);
assert(twice(squared(5)) equals 50);
//assert(squaredAndTwice(10) equals 200);
assert(squared(squared(squared(2))) equals 2^2^2^2);