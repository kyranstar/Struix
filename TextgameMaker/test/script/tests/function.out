11
sup
WowWowWow
false
9
